packages/query-core/src/queryClient.ts

```diff
@@ -113,84 +107,29 @@ export class QueryClient {
   }
 
   getQueryData<TQueryFnData = unknown>(
-    queryKey: QueryKey,
-    filters?: QueryFilters,
+    filters: WithRequired<QueryFilters, 'queryKey'>,
   ): TQueryFnData | undefined {
-    return this.queryCache.find<TQueryFnData>(queryKey, filters)?.state.data
+    return this.queryCache.find<TQueryFnData>(filters)?.state.data
   }
```

Comment:
I wasn't sure about `getQueryData`. I've only ever used it with a queryKey and no other filters. If the key is required, do the other filters even make sense 🤔. Maybe it would be better to have this as just a function that lets you pass in a `queryKey` only?

What do you think @ardeora @DamianOsipiuk ? On the other hand, one object is the consistent way 😅

Reply:
I would opt for either:
- single argument with everything `optional`. As an alias to `getQueryCache().find()`
- single argument of `queryKey`. Then if you need more advanced search you would do `getQueryCache().find()`.

To be honest i would go with only `queryKey` as if i would like to use more filters i would expect to potentially get more than one result. Meaning `getQueriesData`.

Reply:
Yeah I agree with @DamianOsipiuk. For example, if I use the filters like `exact: false` or `predicate`, I'm usually expecting more than one result

Reply:
agree. It's weird that we would only get the first matching result then - that's what we have `getQueriesData` etc. for.

Ok then let's do one param only, that is _just_ the queryKey, and that is required then.

Reply:
sure 🙌 should I apply the same concept on `getQueryState` ? I think it is similar to `getQueryData`

Reply:
yes, they are pretty much the same :)

Reply:
Done ✅ also updated the migration docs to include this change 🙌